internal/blobtesting/map.go

```diff
@@ -192,6 +192,10 @@ func (s *mapStorage) FlushCaches(ctx context.Context) error {
 	return nil
 }
 
+func (s *mapStorage) ExtendBlobRetention(ctx context.Context, b blob.ID, opts blob.ExtendOptions) error {
+	return blob.ErrUnsupportedObjectLock
```

Comment:
nit: perhaps introduce `type UnsupportedBlobRetention struct {}` with default implementation and embed everywhere instead of repeating the same method.